lib/rom/repository/command_proxy.rb

```diff
@@ -11,11 +11,10 @@ class CommandProxy
 
       def initialize(command)
         @command = command
-        @root = Dry::Core::Inflector.singularize(command.name.relation).to_sym
       end
 
       def call(input)
-        command.call(root => input)
+        command.call(command.name.relation => input)
       end
```

Comment:
Right, makes sense not to do something that's not really needed :D

This class is a temporary "solution", the only reason why it exists is because command graphs in rom core, as you noticed, needs the relation name as the key. If we can tweak it so this is no longer need, then this whole proxy can be removed.